src/java/neroxis/map/SCMap.java

```diff
@@ -467,11 +467,17 @@ private void moveObjects(float contentScale, Vector2f offset) {
     }
 
     private void scaleMapContent(float contentScale) {
-        this.biome.getWaterSettings().setElevation(this.biome.getWaterSettings().getElevation() * contentScale);
-        this.biome.getWaterSettings().setElevationDeep(this.biome.getWaterSettings().getElevationDeep() * contentScale);
-        this.biome.getWaterSettings().setElevationAbyss(this.biome.getWaterSettings().getElevationAbyss() * contentScale);
+        float maxHeightScale = getHeightMask(new SymmetrySettings(Symmetry.NONE, Symmetry.NONE, Symmetry.NONE)).getMax() * contentScale;
+        if (maxHeightScale > 255) {
+            maxHeightScale = contentScale / maxHeightScale * 255;
+        } else {
+            maxHeightScale = contentScale;
+        }
```

Comment:
maxHeightScale should be a separate variable as currently it is the maxHeight and then a scale